Lib/collections/__init__.py

```diff
@@ -508,6 +487,10 @@ def __getnewargs__(self):
         '__getnewargs__': __getnewargs__,
         '__match_args__': field_names,
     }
+
+    if classcell is not _nmtuple_classcell_sentinel:
+        class_namespace["__classcell__"] = classcell
```

Comment:
A comment here will help future readers of this code:

```suggestion
    # gh-85795: `super()` calls inside `typing.NamedTuple` methods will not
    # work unless `__classcell__` is propagated by `collections._namedtuple`
    if classcell is not _nmtuple_classcell_sentinel:
        class_namespace["__classcell__"] = classcell
```